src/Work/CompanyPositionRequirements.ts

```diff
@@ -0,0 +1,45 @@
+export interface CompanyPositionRequirements {
+  name: string;
+  nextPosition: string | null;
+  requiredHacking: number;
+  requiredStrength: number;
+  requiredDefense: number;
+  requiredDexterity: number;
+  requiredAgility: number;
+  requiredCharisma: number;
+  requiredReputation: number;
+}
+
+export const newCompanyPositionRequirements = (
+  params?: Partial<CompanyPositionRequirements>,
+): CompanyPositionRequirements => {
+  return {
+    name: params?.name ?? "",
+    nextPosition: params?.nextPosition ?? "",
+    requiredHacking: params?.requiredHacking ?? 0,
+    requiredStrength: params?.requiredStrength ?? 0,
+    requiredDefense: params?.requiredDefense ?? 0,
+    requiredDexterity: params?.requiredDexterity ?? 0,
+    requiredAgility: params?.requiredAgility ?? 0,
+    requiredCharisma: params?.requiredCharisma ?? 0,
+    requiredReputation: params?.requiredReputation ?? 0,
+  };
+};
+
+/** Scale all stats on a CompanyPositionRequirements object by a number. */
+export const scaleCompanyPositionRequirements = (
+  c: CompanyPositionRequirements,
+  n: number,
+): CompanyPositionRequirements => {
+  return {
+    name: c.name,
+    nextPosition: c.nextPosition,
+    requiredHacking: c.requiredHacking > 0 ? c.requiredHacking + n : 0,
+    requiredStrength: c.requiredStrength > 0 ? c.requiredStrength + n : 0,
+    requiredDefense: c.requiredDefense > 0 ? c.requiredDefense + n : 0,
+    requiredDexterity: c.requiredDexterity > 0 ? c.requiredDexterity + n : 0,
+    requiredAgility: c.requiredAgility > 0 ? c.requiredAgility + n : 0,
+    requiredCharisma: c.requiredCharisma > 0 ? c.requiredCharisma + n : 0,
+    requiredReputation: c.requiredReputation,
+  };
+};
```

Comment:
So for something that is being provided to the player for use, I like to think about, how can I shape this so it's most useful to the player. Following the existing shape of the internal code is much less important than giving something the most convenient shape for use by the player. Internal code can be freely reshaped however we want, but changing the shape of stuff provided in the API is very difficult because it is an API break for player scripts.

For instance, we already have the Skills interface, which is what players are given for their own skills. Providing the required skills using the same structure as this existing interface is convenient. e.g. players can check if they meet the skill requirements for a job without needing to hardcode the interface keys by manually comparing their own "hacking" skill value against a "requiredHacking" value and so on for each of the other skills.

So the way I would shape the interface is 
```ts
interface CompanyPositionInfo {
  name: CompanyPosName;
  nextPosition?: CompanyPosName; // could also be a required property with CompanyPosName | null
  requiredReputation: number;
  requiredSkills: Skills;
}
```

Additionally, it looks like this is probably based on WorkStats, which is the only thing in the game with this wonky structure where it's an interface sort of pretending it's a class. Not really a good example to use as a base.

In this case, I would define the interface directly in NetscriptDefinitions (and only there), and if you need it in the Singularity ns file you can import it. For the first implementation I would not use any helper functions and just manually construct the correct shape in getCompanyPositionInfo, later if we decide to reshape some internal code to match the api, we could add a helper function and use it both in the NS function as well as in the internal code for checking job requirements (this could also be done all at the same time, just more work).